code/engine/xrCDB/xrCDB.cpp

```diff
@@ -43,45 +43,12 @@ MODEL::~MODEL() {
     verts_count = 0;
 }
 
-struct BTHREAD_params {
-    MODEL* M;
-    Fvector* V;
-    int Vcnt;
-    TRI* T;
-    int Tcnt;
-    build_callback* BC;
-    void* BCP;
-};
-
-void MODEL::build_thread(void* params) {
-    _initialize_cpu_thread();
-    FPU::m64r();
-    BTHREAD_params P = *((BTHREAD_params*)params);
-    std::lock_guard<std::recursive_mutex> lock(P.M->cs);
-    P.M->build_internal(P.V, P.Vcnt, P.T, P.Tcnt, P.BC, P.BCP);
-    P.M->status = S_READY;
-    // Msg						("* xrCDB: cform build completed, memory usage: %d
-    // K",P.M->memory()/1024);
-}
-
 void MODEL::build(Fvector* V, int Vcnt, TRI* T, int Tcnt, build_callback* bc, void* bcp) {
     R_ASSERT(S_INIT == status);
     R_ASSERT((Vcnt >= 4) && (Tcnt >= 2));
 
-    _initialize_cpu_thread();
-#ifdef _EDITOR
-    build_internal(V, Vcnt, T, Tcnt, bc, bcp);
-#else
-    if (!strstr(Core.Params, "-mt_cdb")) {
-        build_internal(V, Vcnt, T, Tcnt, bc, bcp);
-        status = S_READY;
-    } else {
-        BTHREAD_params P = { this, V, Vcnt, T, Tcnt, bc, bcp };
-        thread_spawn(build_thread, "CDB-construction", 0, &P);
-        while (S_INIT == status)
-            Sleep(5);
-    }
-#endif
+	build_internal(V, Vcnt, T, Tcnt, bc, bcp);
+	status = S_READY;
```

Comment:
You have deleted
```
else
{
    BTHREAD_params P = { this, V, Vcnt, T, Tcnt, bc, bcp };
    thread_spawn(build_thread, "CDB-construction", 0, &P);
    while (S_INIT == status)
        Sleep(5);
}
```
Because it is unefficient? Or what?)

Reply:
This function only runs few times during the program life, I dont see any point of creating thread each time for it, and then sleep and wait until it does something. I might be wrong, but I think there will be no sufficient performance improvement.

Reply:
@Im-dex the existing multi-threading usage is only targeting one thread, it would be better to use parallel loops when needed. It can be put in todo list, but I dont see a good reason to keep gsc implementation for this case.